sos/plugins/block.py

```diff
@@ -22,6 +22,15 @@ class Block(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
 
     plugin_name = 'block'
     profiles = ('storage', 'hardware')
+    option_list = [("all", "get all block info", "fast", False)]
+
+    def get_info_for_all_devices(self):
+        block_devices = [blk for blk in os.listdir('/sys/block/')]
+        for blk in block_devices:
+            all_files = self.call_ext_prog("find /sys/block/%s/ -type f" % blk)
```

Comment:
This has been tried before however, and gets pretty slow on large systems. You may find that using `os.path.walk` and pre-filtering `sysfs` attributes that are unreadable, irrelevant or otherwise not worth collecting gives you better results.